{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "876ccb32-62dd-4029-a3f0-e2a9fa058763",
   "metadata": {},
   "source": [
    "Dans Launcher terminal bash\n",
    "se placer dans directory Projet-Methylation\n",
    "cd /shared/projects/202304_duo/sletuve/Projet/Projet-Methylation\n",
    "vérifier avec pwd\n",
    "\n",
    "# Dépôt GitHub:\n",
    "**création d'un dépôt**\n",
    "\n",
    "Sur Github, en haut à droite, cliquer sur +\n",
    "\n",
    "Puis nommer le dépôt (ici Projet-Methylation)\n",
    "\n",
    "git@github.com:sevlet94/Projet-Methylation.git\n",
    "\n",
    "et clone à partir de l'onglet 'Git' de Jypyter dans le dossier\n",
    "\n",
    "se déplacer dans le dossier cloné: cd Projet-Methylation\n",
    "\n",
    "**configuration du dépot local**\n",
    "\n",
    "$ git config --global user.name \"Séverine Létuvé\"\n",
    "\n",
    "$ git config --global user.email \"severine.letuve@inserm.fr\"\n",
    "\n",
    "vérification avec $ git config --list | grep user\n",
    "\n",
    "**configuration de Nano comme éditeur par défaut**\n",
    "\n",
    "$ git config --global core.editor \"nano\"\n",
    "\n",
    "Contenu du dossier\n",
    "$ git status\n",
    "\n",
    "**Ajout de NoteBookAnalyseMethylation.ipynb**\n",
    "$ git add NoteBookAnalyseMethylation.ipynb\n",
    "$ git commit -m \"Premier commit\"\n",
    "$ git push\n",
    "\n",
    "**Création README**\n",
    "Sur site Github\n",
    "puis récupérer sur local\n",
    "\n",
    "$ pull\n",
    "\n",
    "Lire contenu README.md\n",
    "$ cat README.md\n",
    "\n",
    "Compte-rendu de l'activité sur dépôt\n",
    "$ git log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "79d64273-fc47-4083-9a65-7222cf4e96ea",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "The minfi package is described in:\n",
       "\n",
       "Aryee MJ, Jaffe AE, Corrada-Bravo H, Ladd-Acosta C, Feinberg AP, Hansen\n",
       "KD, Irizarry RA (2014). “Minfi: A flexible and comprehensive\n",
       "Bioconductor package for the analysis of Infinium DNA Methylation\n",
       "microarrays.” _Bioinformatics_, *30*(10), 1363-1369. doi:\n",
       "10.1093/bioinformatics/btu049 (URL:\n",
       "https://doi.org/10.1093/bioinformatics/btu049).\n",
       "\n",
       "SWAN normalization (preprocessSWAN) is described in:\n",
       "\n",
       "Maksimovic J, Gordon L, Oshlack A (2012). “SWAN: Subset quantile\n",
       "Within-Array Normalization for Illumina Infinium HumanMethylation450\n",
       "BeadChips.” _Genome Biology_, *13*(6), R44. doi:\n",
       "10.1186/gb-2012-13-6-r44 (URL:\n",
       "https://doi.org/10.1186/gb-2012-13-6-r44).\n",
       "\n",
       "Funnorm normalization (preprocesFunnorm) is described in:\n",
       "\n",
       "Fortin J, Labbe A, Lemire M, Zanke BW, Hudson TJ, Fertig EJ, Greenwood\n",
       "CM, Hansen KD (2014). “Functional normalization of 450k methylation\n",
       "array data improves replication in large cancer studies.” _Genome\n",
       "Biology_, *15*(12), 503. doi: 10.1186/s13059-014-0503-2 (URL:\n",
       "https://doi.org/10.1186/s13059-014-0503-2).\n",
       "\n",
       "noob background correction (preprocessNoob) is described in:\n",
       "\n",
       "Triche TJ, Weisenberger DJ, Van Den Berg D, Laird PW, Siegmund KD\n",
       "(2013). “Low-level processing of Illumina Infinium DNA Methylation\n",
       "BeadArrays.” _Nucleic Acids Research_, *41*(7), e90. doi:\n",
       "10.1093/nar/gkt090 (URL: https://doi.org/10.1093/nar/gkt090).\n",
       "\n",
       "Reconstruction of A/B compartments (compartments, extractAB) is\n",
       "described in:\n",
       "\n",
       "Fortin J, Hansen KD (2015). “Reconstructing A/B compartments as\n",
       "revealed by Hi-C using long-range correlations in epigenetic data.”\n",
       "_Genome Biology_, *16*, 180. doi: 10.1186/s13059-015-0741-y (URL:\n",
       "https://doi.org/10.1186/s13059-015-0741-y).\n",
       "\n",
       "Gap hunting is described in:\n",
       "\n",
       "Andrews SV, Ladd-Acosta C, Feinberg AP, Hansen KD, Fallin MD (2016).\n",
       "“'Gap hunting' to characterize clustered probe signals in Illumina\n",
       "methylation array data.” _Epigenetics & Chromatin_, *9*, 56. doi:\n",
       "10.1186/s13072-016-0107-z (URL:\n",
       "https://doi.org/10.1186/s13072-016-0107-z).\n",
       "\n",
       "Extending minfi to support EPIC arrays is described in:\n",
       "\n",
       "Fortin J, Triche TJ, Hansen KD (2017). “Preprocessing, normalization\n",
       "and integration of the Illumina HumanMethylationEPIC array with minfi.”\n",
       "_Bioinformatics_, *33*(4). doi: 10.1093/bioinformatics/btw691 (URL:\n",
       "https://doi.org/10.1093/bioinformatics/btw691).\n",
       "\n",
       "To see these entries in BibTeX format, use 'print(<citation>,\n",
       "bibtex=TRUE)', 'toBibtex(.)', or set\n",
       "'options(citation.bibtex.max=999)'.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "citation(\"minfi\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9f37f1c-8243-491f-8132-68f7650de05a",
   "metadata": {},
   "source": [
    "# Utilisation du package Minfi pour l'analyse de puces à méthylation 450K Illumina (Illumina’s Methylation arrays 450K and Epic)\n",
    "\n",
    "\n",
    "\n",
    "**User's guide**\n",
    "https://bioconductor.org/packages/devel/bioc/vignettes/minfi/inst/doc/minfi.html\n",
    "input data to this package are IDAT files\n",
    "\n",
    "***Principe du methylation array***\n",
    "450,000 CpG positions tested (out of 28 millions CpG)/ Each CpG is associated with two measurements: a methylated measurement and an “un”-methylated measurement\n",
    "\n",
    "*These two values can be measured in one of two ways: using a “Type I” design or a “Type II design”. CpGs measured using a Type I design are measured using a single color, with two different probes in the same color channel providing the methylated and the unmethylated measurements. CpGs measured using a Type II design are measured using a single probe, and two different colors provide the methylated and the unmethylated measurements.\n",
    "Practically, this implies that on this array there is not a one-to-one correspondence between probes and CpG positions. We have therefore tried to be precise about this and we refer to a “locus” (or “CpG”) when we refer to a single-base genomic locus, and we differentiate this from a “probe”. The previous generation 27k methylation array uses only the Type I design, and the EPIC arrays uses both Type I and Type II.\n",
    "Differences in DNA methylation between samples can either be at a single CpG which is called a differentially methylated position (DMP), or at a regional level which is called a differentially methylated region (DMR)*\n",
    "\n",
    "\n",
    " \n",
    "**Installation**\n",
    "To install core packages, type the following in an R command window:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b3bdea7f-6b81-4527-9080-203eca146098",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘BiocManager’ was built under R version 4.0.5”\n",
      "Bioconductor version '3.12' is out-of-date; the current release version '3.18'\n",
      "  is available with R version '4.3'; see https://bioconductor.org/install\n",
      "\n",
      "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'?repositories' for details\n",
      "\n",
      "replacement repositories:\n",
      "    CRAN: https://cran.r-project.org\n",
      "\n",
      "\n",
      "Bioconductor version 3.12 (BiocManager 1.30.16), R 4.0.3 (2020-10-10)\n",
      "\n",
      "Installation paths not writeable, unable to update packages\n",
      "  path: /shared/ifbstor1/software/miniconda/envs/r-4.0.3/lib/R/library\n",
      "  packages:\n",
      "    abcrf, acepack, ade4, adegenet, admisc, amap, apcluster, ape, argparse,\n",
      "    aricode, ashr, askpass, babelgene, backports, base64, BAT, batchtools,\n",
      "    bayesm, bayestestR, bbmle, bdsmatrix, beanplot, BH, BiasedUrn, BiocManager,\n",
      "    bit, blob, bookdown, boot, brew, brio, broom, broom.helpers, bslib, cachem,\n",
      "    Cairo, callr, car, carData, caret, changepoint, checkmate, circlize, class,\n",
      "    classInt, cli, clipr, clue, cluster, codetools, colorspace, colourpicker,\n",
      "    commonmark, ComplexUpset, compositions, conquer, countrycode, covr, cpp11,\n",
      "    crayon, credentials, curl, dartR, data.table, datawizard, DBI, dbplyr,\n",
      "    deldir, dendextend, densityClust, DEoptimR, desc, devtools, dichromat,\n",
      "    digest, dismo, doMC, doParallel, doRNG, doSNOW, dplyr, dqrng, DT, dtplyr,\n",
      "    e1071, effectsize, ellipse, emdbook, evaluate, expm, extrafont, FactoMineR,\n",
      "    fansi, farver, fastmap, fastmatch, FD, findpython, fitdistrplus, float,\n",
      "    FNN, fontawesome, forcats, foreach, foreign, formatR, Formula, fs, furrr,\n",
      "    future, future.apply, gap, gargle, gaston, gdata, gdistance, gdtools,\n",
      "    generics, GenSA, geometry, gert, getopt, ggbeeswarm, ggdendro, ggfittext,\n",
      "    ggforce, gggenes, ggiraph, ggplot2, ggpubr, ggraph, ggrastr, ggrepel,\n",
      "    ggridges, ggsci, ggsignif, ggvenn, ggVennDiagram, gh, git2r, gitcreds,\n",
      "    glmnet, globals, gmm, gmodels, gmp, googledrive, googlesheets4, gower,\n",
      "    gplots, gprofiler2, graphlayouts, grImport, gt, gtable, gtools, gtsummary,\n",
      "    hardhat, HardyWeinberg, haven, heatmaply, hexbin, hierfstat, highr,\n",
      "    hitandrun, Hmisc, hms, htmltools, htmlwidgets, httpuv, httr, hwriter,\n",
      "    hypervolume, ica, igraph, imputeLCMD, insight, ipred, IRdisplay, IRkernel,\n",
      "    irlba, isoband, iterators, JADE, jpeg, jsonlite, kernlab, KernSmooth,\n",
      "    knitr, kohonen, ks, kSamples, labdsv, labeling, labelled, later, lattice,\n",
      "    latticeExtra, lava, LDRTools, learnr, leiden, lgr, LiblineaR, lifecycle,\n",
      "    linprog, listenv, lme4, lmtest, lpSolve, lsa, lubridate, magic, MALDIquant,\n",
      "    maps, markdown, MASS, Matrix, MatrixModels, matrixStats, mclust, memoise,\n",
      "    memuse, mgcv, mice, minqa, mixKernel, mixsqp, mixtools, mlapi, mnormt,\n",
      "    modelr, moments, msigdbr, mvtnorm, narray, ncdf4, nleqslv, nlme, nloptr,\n",
      "    nls2, NMF, nnet, nnls, norm, oompaData, openssl, openxlsx, optparse,\n",
      "    packrat, palmerpenguins, parallelDist, parallelly, parameters, parsedate,\n",
      "    parsnip, patchwork, pbapply, pbdZMQ, pbmcapply, pdist, pegas, performance,\n",
      "    permute, pillar, pingr, pkgbuild, pkgcache, pkgload, pkgmaker, plotly,\n",
      "    plotrix, plyr, png, polyclip, polynom, PopGenReport, pracma, prettyunits,\n",
      "    pROC, processx, prodlim, progressr, proj4, promises, proxy, ps, psych,\n",
      "    purrr, qap, qqman, quantreg, R.devices, R.methodsS3, R.oo, R.utils, ragg,\n",
      "    ranger, raster, rbibutils, rcdd, RCircos, RColorBrewer, Rcpp, RcppAnnoy,\n",
      "    RcppArmadillo, RcppEigen, RcppHNSW, RcppNumerical, RcppParallel, RCurl,\n",
      "    Rdpack, readr, readxl, recipes, rematch, remotes, renv, repr, reprex,\n",
      "    reshape, reticulate, rex, rgl, RgoogleMaps, RhpcBLASctl, rintrojs, rio,\n",
      "    rjags, rJava, rjson, RJSONIO, Rlab, rlang, RMariaDB, rmarkdown, Rmpfr,\n",
      "    robustbase, roxygen2, rpart, rpart.plot, rprojroot, rsample, rsparse,\n",
      "    RSpectra, RSQLite, rstatix, rstudioapi, rsvg, Rtsne, Rttf2pt1, rversions,\n",
      "    rvest, s2, sandwich, sass, scales, scattermore, scatterplot3d, sctransform,\n",
      "    segmented, seqinr, seqminer, seriation, sessioninfo, Seurat, SeuratObject,\n",
      "    sf, shadowtext, shiny, shinyAce, shinyalert, shinyBS, shinyFiles, shinyjs,\n",
      "    shinytest, shinyWidgets, Signac, slam, slider, sm, SnowballC, sourcetools,\n",
      "    sp, sparsesvd, spatstat, spatstat.data, spatstat.geom, spatstat.linnet,\n",
      "    spatstat.sparse, spatstat.utils, spData, spdep, StatMatch, statmod,\n",
      "    stringi, stringr, SuppDists, survey, survival, svglite, sys, systemfonts,\n",
      "    terra, testthat, text2vec, textshaping, TFMPvalue, tibble, tictoc,\n",
      "    tidygraph, tidyr, tidyselect, tidyverse, timeDate, tinytex, tm, tmvtnorm,\n",
      "    treemap, truncnorm, TSP, TTR, tweenr, tzdb, umap, units, usethis, utf8,\n",
      "    uuid, uwot, V8, vcd, vcfR, vctrs, vegan, venn, VennDiagram, VGAM, VIM,\n",
      "    vioplot, viridis, viridisLite, vroom, waiter, waldo, warp, webshot, WGCNA,\n",
      "    whisker, withr, wk, xfun, xgboost, XML, xml2, xts, yaml, yardstick,\n",
      "    yulab.utils, zip, zoo\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if (!require(\"BiocManager\", quietly = TRUE))\n",
    "    install.packages(\"BiocManager\")\n",
    "BiocManager::install(version = \"3.12\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f711986b-6414-468c-86c2-a3ad88a8807b",
   "metadata": {},
   "source": [
    "To install minfi package, start R (version \"4.0.3\") and enter:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ae649743-9b71-4c42-9b90-abc8baad5665",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'?repositories' for details\n",
      "\n",
      "replacement repositories:\n",
      "    CRAN: https://cran.r-project.org\n",
      "\n",
      "\n",
      "Bioconductor version 3.12 (BiocManager 1.30.16), R 4.0.3 (2020-10-10)\n",
      "\n",
      "Warning message:\n",
      "“package(s) not installed when version(s) same as current; use `force = TRUE` to\n",
      "  re-install: 'minfi'”\n",
      "Installation paths not writeable, unable to update packages\n",
      "  path: /shared/ifbstor1/software/miniconda/envs/r-4.0.3/lib/R/library\n",
      "  packages:\n",
      "    abcrf, acepack, ade4, adegenet, admisc, amap, apcluster, ape, argparse,\n",
      "    aricode, ashr, askpass, babelgene, backports, base64, BAT, batchtools,\n",
      "    bayesm, bayestestR, bbmle, bdsmatrix, beanplot, BH, BiasedUrn, BiocManager,\n",
      "    bit, blob, bookdown, boot, brew, brio, broom, broom.helpers, bslib, cachem,\n",
      "    Cairo, callr, car, carData, caret, changepoint, checkmate, circlize, class,\n",
      "    classInt, cli, clipr, clue, cluster, codetools, colorspace, colourpicker,\n",
      "    commonmark, ComplexUpset, compositions, conquer, countrycode, covr, cpp11,\n",
      "    crayon, credentials, curl, dartR, data.table, datawizard, DBI, dbplyr,\n",
      "    deldir, dendextend, densityClust, DEoptimR, desc, devtools, dichromat,\n",
      "    digest, dismo, doMC, doParallel, doRNG, doSNOW, dplyr, dqrng, DT, dtplyr,\n",
      "    e1071, effectsize, ellipse, emdbook, evaluate, expm, extrafont, FactoMineR,\n",
      "    fansi, farver, fastmap, fastmatch, FD, findpython, fitdistrplus, float,\n",
      "    FNN, fontawesome, forcats, foreach, foreign, formatR, Formula, fs, furrr,\n",
      "    future, future.apply, gap, gargle, gaston, gdata, gdistance, gdtools,\n",
      "    generics, GenSA, geometry, gert, getopt, ggbeeswarm, ggdendro, ggfittext,\n",
      "    ggforce, gggenes, ggiraph, ggplot2, ggpubr, ggraph, ggrastr, ggrepel,\n",
      "    ggridges, ggsci, ggsignif, ggvenn, ggVennDiagram, gh, git2r, gitcreds,\n",
      "    glmnet, globals, gmm, gmodels, gmp, googledrive, googlesheets4, gower,\n",
      "    gplots, gprofiler2, graphlayouts, grImport, gt, gtable, gtools, gtsummary,\n",
      "    hardhat, HardyWeinberg, haven, heatmaply, hexbin, hierfstat, highr,\n",
      "    hitandrun, Hmisc, hms, htmltools, htmlwidgets, httpuv, httr, hwriter,\n",
      "    hypervolume, ica, igraph, imputeLCMD, insight, ipred, IRdisplay, IRkernel,\n",
      "    irlba, isoband, iterators, JADE, jpeg, jsonlite, kernlab, KernSmooth,\n",
      "    knitr, kohonen, ks, kSamples, labdsv, labeling, labelled, later, lattice,\n",
      "    latticeExtra, lava, LDRTools, learnr, leiden, lgr, LiblineaR, lifecycle,\n",
      "    linprog, listenv, lme4, lmtest, lpSolve, lsa, lubridate, magic, MALDIquant,\n",
      "    maps, markdown, MASS, Matrix, MatrixModels, matrixStats, mclust, memoise,\n",
      "    memuse, mgcv, mice, minqa, mixKernel, mixsqp, mixtools, mlapi, mnormt,\n",
      "    modelr, moments, msigdbr, mvtnorm, narray, ncdf4, nleqslv, nlme, nloptr,\n",
      "    nls2, NMF, nnet, nnls, norm, oompaData, openssl, openxlsx, optparse,\n",
      "    packrat, palmerpenguins, parallelDist, parallelly, parameters, parsedate,\n",
      "    parsnip, patchwork, pbapply, pbdZMQ, pbmcapply, pdist, pegas, performance,\n",
      "    permute, pillar, pingr, pkgbuild, pkgcache, pkgload, pkgmaker, plotly,\n",
      "    plotrix, plyr, png, polyclip, polynom, PopGenReport, pracma, prettyunits,\n",
      "    pROC, processx, prodlim, progressr, proj4, promises, proxy, ps, psych,\n",
      "    purrr, qap, qqman, quantreg, R.devices, R.methodsS3, R.oo, R.utils, ragg,\n",
      "    ranger, raster, rbibutils, rcdd, RCircos, RColorBrewer, Rcpp, RcppAnnoy,\n",
      "    RcppArmadillo, RcppEigen, RcppHNSW, RcppNumerical, RcppParallel, RCurl,\n",
      "    Rdpack, readr, readxl, recipes, rematch, remotes, renv, repr, reprex,\n",
      "    reshape, reticulate, rex, rgl, RgoogleMaps, RhpcBLASctl, rintrojs, rio,\n",
      "    rjags, rJava, rjson, RJSONIO, Rlab, rlang, RMariaDB, rmarkdown, Rmpfr,\n",
      "    robustbase, roxygen2, rpart, rpart.plot, rprojroot, rsample, rsparse,\n",
      "    RSpectra, RSQLite, rstatix, rstudioapi, rsvg, Rtsne, Rttf2pt1, rversions,\n",
      "    rvest, s2, sandwich, sass, scales, scattermore, scatterplot3d, sctransform,\n",
      "    segmented, seqinr, seqminer, seriation, sessioninfo, Seurat, SeuratObject,\n",
      "    sf, shadowtext, shiny, shinyAce, shinyalert, shinyBS, shinyFiles, shinyjs,\n",
      "    shinytest, shinyWidgets, Signac, slam, slider, sm, SnowballC, sourcetools,\n",
      "    sp, sparsesvd, spatstat, spatstat.data, spatstat.geom, spatstat.linnet,\n",
      "    spatstat.sparse, spatstat.utils, spData, spdep, StatMatch, statmod,\n",
      "    stringi, stringr, SuppDists, survey, survival, svglite, sys, systemfonts,\n",
      "    terra, testthat, text2vec, textshaping, TFMPvalue, tibble, tictoc,\n",
      "    tidygraph, tidyr, tidyselect, tidyverse, timeDate, tinytex, tm, tmvtnorm,\n",
      "    treemap, truncnorm, TSP, TTR, tweenr, tzdb, umap, units, usethis, utf8,\n",
      "    uuid, uwot, V8, vcd, vcfR, vctrs, vegan, venn, VennDiagram, VGAM, VIM,\n",
      "    vioplot, viridis, viridisLite, vroom, waiter, waldo, warp, webshot, WGCNA,\n",
      "    whisker, withr, wk, xfun, xgboost, XML, xml2, xts, yaml, yardstick,\n",
      "    yulab.utils, zip, zoo\n",
      "\n"
     ]
    }
   ],
   "source": [
    "BiocManager::install(\"minfi\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ccdcae67-0c9a-4e9f-943f-987e6f1ce4d0",
   "metadata": {},
   "source": [
    "R sur JupyterLab vs.Rstudio?\n",
    "Je veux utiliser le package Minfi pour analyser des données de Methylation array.\n",
    "Il semblerait que j'aie besoin d'installer préalablement Bioconductor.\n",
    "Pour R4.0.3(via JupyterLab)= version 3.12\n",
    "Pour R4.1.1 (via Rstudio)= version 3.18\n",
    "Quand j'essaie d'installer le package Minfi via BiocManager en utilisant un notebook Jupyter, j'obtiens un message d'erreur car version trop ancienne de plusieurs packages nécessaires =  ne peuvent plus être updatés.\n",
    "Par contre, je peux loader les versions plus récentes de ces packages via BiocManager sur RStudio.\n",
    "**Aide trouvée sur uilitR** \n",
    "*4.2.5 Mettre à jour les packages*\n",
    "Il peut arriver qu’un de vos programmes génère une erreur parce que vous utilisez un package dans une version trop ancienne. Pour mettre à jour un package, il suffit de le réinstaller avec **install.packages**. Cette méthode de mise à jour est la plus simple, et résout la très grande majorité des problèmes de version de packages.\n",
    "Il existe également la fonction **update.packages**. Cette fonction ne sert pas à mettre un package particulier, mais à mettre à jour tous les packages d’une librairie. Vous pouvez utiliser cette fonction si vous le souhaitez, mais son usage est plus complexe que celui d’install.packages. Vous pouvez en apprendre davantage sur cette fonction en consultant l’aide associée (?update.packages).\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7df9c1c6-0645-4f11-9b3d-1bf5eb7febc3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'?repositories' for details\n",
      "\n",
      "replacement repositories:\n",
      "    CRAN: https://cran.r-project.org\n",
      "\n",
      "\n",
      "Bioconductor version 3.12 (BiocManager 1.30.16), R 4.0.3 (2020-10-10)\n",
      "\n",
      "Warning message:\n",
      "“package(s) not installed when version(s) same as current; use `force = TRUE` to\n",
      "  re-install: 'minfi'”\n",
      "Installation paths not writeable, unable to update packages\n",
      "  path: /shared/ifbstor1/software/miniconda/envs/r-4.0.3/lib/R/library\n",
      "  packages:\n",
      "    abcrf, acepack, ade4, adegenet, admisc, amap, apcluster, ape, argparse,\n",
      "    aricode, ashr, askpass, babelgene, backports, base64, BAT, batchtools,\n",
      "    bayesm, bayestestR, bbmle, bdsmatrix, beanplot, BH, BiasedUrn, BiocManager,\n",
      "    bit, blob, bookdown, boot, brew, brio, broom, broom.helpers, bslib, cachem,\n",
      "    Cairo, callr, car, carData, caret, changepoint, checkmate, circlize, class,\n",
      "    classInt, cli, clipr, clue, cluster, codetools, colorspace, colourpicker,\n",
      "    commonmark, ComplexUpset, compositions, conquer, countrycode, covr, cpp11,\n",
      "    crayon, credentials, curl, dartR, data.table, datawizard, DBI, dbplyr,\n",
      "    deldir, dendextend, densityClust, DEoptimR, desc, devtools, dichromat,\n",
      "    digest, dismo, doMC, doParallel, doRNG, doSNOW, dplyr, dqrng, DT, dtplyr,\n",
      "    e1071, effectsize, ellipse, emdbook, evaluate, expm, extrafont, FactoMineR,\n",
      "    fansi, farver, fastmap, fastmatch, FD, findpython, fitdistrplus, float,\n",
      "    FNN, fontawesome, forcats, foreach, foreign, formatR, Formula, fs, furrr,\n",
      "    future, future.apply, gap, gargle, gaston, gdata, gdistance, gdtools,\n",
      "    generics, GenSA, geometry, gert, getopt, ggbeeswarm, ggdendro, ggfittext,\n",
      "    ggforce, gggenes, ggiraph, ggplot2, ggpubr, ggraph, ggrastr, ggrepel,\n",
      "    ggridges, ggsci, ggsignif, ggvenn, ggVennDiagram, gh, git2r, gitcreds,\n",
      "    glmnet, globals, gmm, gmodels, gmp, googledrive, googlesheets4, gower,\n",
      "    gplots, gprofiler2, graphlayouts, grImport, gt, gtable, gtools, gtsummary,\n",
      "    hardhat, HardyWeinberg, haven, heatmaply, hexbin, hierfstat, highr,\n",
      "    hitandrun, Hmisc, hms, htmltools, htmlwidgets, httpuv, httr, hwriter,\n",
      "    hypervolume, ica, igraph, imputeLCMD, insight, ipred, IRdisplay, IRkernel,\n",
      "    irlba, isoband, iterators, JADE, jpeg, jsonlite, kernlab, KernSmooth,\n",
      "    knitr, kohonen, ks, kSamples, labdsv, labeling, labelled, later, lattice,\n",
      "    latticeExtra, lava, LDRTools, learnr, leiden, lgr, LiblineaR, lifecycle,\n",
      "    linprog, listenv, lme4, lmtest, lpSolve, lsa, lubridate, magic, MALDIquant,\n",
      "    maps, markdown, MASS, Matrix, MatrixModels, matrixStats, mclust, memoise,\n",
      "    memuse, mgcv, mice, minqa, mixKernel, mixsqp, mixtools, mlapi, mnormt,\n",
      "    modelr, moments, msigdbr, mvtnorm, narray, ncdf4, nleqslv, nlme, nloptr,\n",
      "    nls2, NMF, nnet, nnls, norm, oompaData, openssl, openxlsx, optparse,\n",
      "    packrat, palmerpenguins, parallelDist, parallelly, parameters, parsedate,\n",
      "    parsnip, patchwork, pbapply, pbdZMQ, pbmcapply, pdist, pegas, performance,\n",
      "    permute, pillar, pingr, pkgbuild, pkgcache, pkgload, pkgmaker, plotly,\n",
      "    plotrix, plyr, png, polyclip, polynom, PopGenReport, pracma, prettyunits,\n",
      "    pROC, processx, prodlim, progressr, proj4, promises, proxy, ps, psych,\n",
      "    purrr, qap, qqman, quantreg, R.devices, R.methodsS3, R.oo, R.utils, ragg,\n",
      "    ranger, raster, rbibutils, rcdd, RCircos, RColorBrewer, Rcpp, RcppAnnoy,\n",
      "    RcppArmadillo, RcppEigen, RcppHNSW, RcppNumerical, RcppParallel, RCurl,\n",
      "    Rdpack, readr, readxl, recipes, rematch, remotes, renv, repr, reprex,\n",
      "    reshape, reticulate, rex, rgl, RgoogleMaps, RhpcBLASctl, rintrojs, rio,\n",
      "    rjags, rJava, rjson, RJSONIO, Rlab, rlang, RMariaDB, rmarkdown, Rmpfr,\n",
      "    robustbase, roxygen2, rpart, rpart.plot, rprojroot, rsample, rsparse,\n",
      "    RSpectra, RSQLite, rstatix, rstudioapi, rsvg, Rtsne, Rttf2pt1, rversions,\n",
      "    rvest, s2, sandwich, sass, scales, scattermore, scatterplot3d, sctransform,\n",
      "    segmented, seqinr, seqminer, seriation, sessioninfo, Seurat, SeuratObject,\n",
      "    sf, shadowtext, shiny, shinyAce, shinyalert, shinyBS, shinyFiles, shinyjs,\n",
      "    shinytest, shinyWidgets, Signac, slam, slider, sm, SnowballC, sourcetools,\n",
      "    sp, sparsesvd, spatstat, spatstat.data, spatstat.geom, spatstat.linnet,\n",
      "    spatstat.sparse, spatstat.utils, spData, spdep, StatMatch, statmod,\n",
      "    stringi, stringr, SuppDists, survey, survival, svglite, sys, systemfonts,\n",
      "    terra, testthat, text2vec, textshaping, TFMPvalue, tibble, tictoc,\n",
      "    tidygraph, tidyr, tidyselect, tidyverse, timeDate, tinytex, tm, tmvtnorm,\n",
      "    treemap, truncnorm, TSP, TTR, tweenr, tzdb, umap, units, usethis, utf8,\n",
      "    uuid, uwot, V8, vcd, vcfR, vctrs, vegan, venn, VennDiagram, VGAM, VIM,\n",
      "    vioplot, viridis, viridisLite, vroom, waiter, waldo, warp, webshot, WGCNA,\n",
      "    whisker, withr, wk, xfun, xgboost, XML, xml2, xts, yaml, yardstick,\n",
      "    yulab.utils, zip, zoo\n",
      "\n"
     ]
    }
   ],
   "source": [
    "BiocManager::install(\"minfi\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4f4a740f-488b-4278-92dd-80ad8f918226",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre class=language-r><code>function (lib.loc = NULL, repos = getOption(\"repos\"), contriburl = contrib.url(repos, \n",
       "<span style=white-space:pre-wrap>    type), method, instlib = NULL, ask = TRUE, available = NULL, </span>\n",
       "<span style=white-space:pre-wrap>    oldPkgs = NULL, ..., checkBuilt = FALSE, type = getOption(\"pkgType\")) </span>\n",
       "{\n",
       "<span style=white-space:pre-wrap>    force(ask)</span>\n",
       "<span style=white-space:pre-wrap>    text.select &lt;- function(old) {</span>\n",
       "<span style=white-space:pre-wrap>        update &lt;- NULL</span>\n",
       "<span style=white-space:pre-wrap>        for (k in seq_len(nrow(old))) {</span>\n",
       "<span style=white-space:pre-wrap>            cat(old[k, \"Package\"], \":\\n\", \"Version\", old[k, \"Installed\"], </span>\n",
       "<span style=white-space:pre-wrap>                \"installed in\", old[k, \"LibPath\"], if (checkBuilt) </span>\n",
       "<span style=white-space:pre-wrap>                  paste(\"built under R\", old[k, \"Built\"]), \"\\n\", </span>\n",
       "<span style=white-space:pre-wrap>                \"Version\", old[k, \"ReposVer\"], \"available at\", </span>\n",
       "<span style=white-space:pre-wrap>                simplifyRepos(old[k, \"Repository\"], type))</span>\n",
       "<span style=white-space:pre-wrap>            cat(\"\\n\")</span>\n",
       "<span style=white-space:pre-wrap>            answer &lt;- askYesNo(\"Update?\")</span>\n",
       "<span style=white-space:pre-wrap>            if (is.na(answer)) {</span>\n",
       "<span style=white-space:pre-wrap>                cat(\"cancelled by user\\n\")</span>\n",
       "<span style=white-space:pre-wrap>                return(invisible())</span>\n",
       "<span style=white-space:pre-wrap>            }</span>\n",
       "<span style=white-space:pre-wrap>            if (isTRUE(answer)) </span>\n",
       "<span style=white-space:pre-wrap>                update &lt;- rbind(update, old[k, ])</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        update</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    if (is.null(lib.loc)) </span>\n",
       "<span style=white-space:pre-wrap>        lib.loc &lt;- .libPaths()</span>\n",
       "<span style=white-space:pre-wrap>    if (type == \"both\" &amp;&amp; (!missing(contriburl) || !is.null(available))) {</span>\n",
       "<span style=white-space:pre-wrap>        stop(\"specifying 'contriburl' or 'available' requires a single type, not type = \\\"both\\\"\")</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    if (is.null(available)) {</span>\n",
       "<span style=white-space:pre-wrap>        available &lt;- available.packages(contriburl = contriburl, </span>\n",
       "<span style=white-space:pre-wrap>            method = method, ...)</span>\n",
       "<span style=white-space:pre-wrap>        if (missing(repos)) </span>\n",
       "<span style=white-space:pre-wrap>            repos &lt;- getOption(\"repos\")</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    if (!is.matrix(oldPkgs) &amp;&amp; is.character(oldPkgs)) {</span>\n",
       "<span style=white-space:pre-wrap>        subset &lt;- oldPkgs</span>\n",
       "<span style=white-space:pre-wrap>        oldPkgs &lt;- NULL</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    else subset &lt;- NULL</span>\n",
       "<span style=white-space:pre-wrap>    if (is.null(oldPkgs)) {</span>\n",
       "<span style=white-space:pre-wrap>        oldPkgs &lt;- old.packages(lib.loc = lib.loc, contriburl = contriburl, </span>\n",
       "<span style=white-space:pre-wrap>            method = method, available = available, checkBuilt = checkBuilt)</span>\n",
       "<span style=white-space:pre-wrap>        if (missing(repos)) </span>\n",
       "<span style=white-space:pre-wrap>            repos &lt;- getOption(\"repos\")</span>\n",
       "<span style=white-space:pre-wrap>        if (!is.null(oldPkgs)) {</span>\n",
       "<span style=white-space:pre-wrap>            pkg &lt;- 0L</span>\n",
       "<span style=white-space:pre-wrap>            while (pkg &lt; nrow(oldPkgs)) {</span>\n",
       "<span style=white-space:pre-wrap>                pkg &lt;- pkg + 1L</span>\n",
       "<span style=white-space:pre-wrap>                if (find.package(oldPkgs[pkg], lib.loc = lib.loc) != </span>\n",
       "<span style=white-space:pre-wrap>                  find.package(oldPkgs[pkg], lib.loc = oldPkgs[pkg, </span>\n",
       "<span style=white-space:pre-wrap>                    2])) {</span>\n",
       "<span style=white-space:pre-wrap>                  warning(sprintf(\"package '%s' in library '%s' will not be updated\", </span>\n",
       "<span style=white-space:pre-wrap>                    oldPkgs[pkg], oldPkgs[pkg, 2]), call. = FALSE, </span>\n",
       "<span style=white-space:pre-wrap>                    immediate. = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>                  oldPkgs &lt;- oldPkgs[-pkg, , drop = FALSE]</span>\n",
       "<span style=white-space:pre-wrap>                  pkg &lt;- pkg - 1L</span>\n",
       "<span style=white-space:pre-wrap>                }</span>\n",
       "<span style=white-space:pre-wrap>            }</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(oldPkgs)) </span>\n",
       "<span style=white-space:pre-wrap>            return(invisible())</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    else if (!(is.matrix(oldPkgs) &amp;&amp; is.character(oldPkgs))) </span>\n",
       "<span style=white-space:pre-wrap>        stop(\"invalid 'oldPkgs'; must be a character vector or a result from old.packages()\")</span>\n",
       "<span style=white-space:pre-wrap>    if (!is.null(subset)) {</span>\n",
       "<span style=white-space:pre-wrap>        oldPkgs &lt;- oldPkgs[rownames(oldPkgs) %in% subset, , drop = FALSE]</span>\n",
       "<span style=white-space:pre-wrap>        if (nrow(oldPkgs) == 0) </span>\n",
       "<span style=white-space:pre-wrap>            return(invisible())</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    update &lt;- if (is.character(ask) &amp;&amp; ask == \"graphics\") {</span>\n",
       "<span style=white-space:pre-wrap>        if (.Platform$OS.type == \"windows\" || .Platform$GUI == </span>\n",
       "<span style=white-space:pre-wrap>            \"AQUA\" || (capabilities(\"tcltk\") &amp;&amp; capabilities(\"X11\"))) {</span>\n",
       "<span style=white-space:pre-wrap>            k &lt;- select.list(oldPkgs[, 1L], oldPkgs[, 1L], multiple = TRUE, </span>\n",
       "<span style=white-space:pre-wrap>                title = \"Packages to be updated\", graphics = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>            oldPkgs[match(k, oldPkgs[, 1L]), , drop = FALSE]</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        else text.select(oldPkgs)</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    else if (isTRUE(ask)) </span>\n",
       "<span style=white-space:pre-wrap>        text.select(oldPkgs)</span>\n",
       "<span style=white-space:pre-wrap>    else oldPkgs</span>\n",
       "<span style=white-space:pre-wrap>    if (length(update)) {</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(instlib)) </span>\n",
       "<span style=white-space:pre-wrap>            instlib &lt;- update[, \"LibPath\"]</span>\n",
       "<span style=white-space:pre-wrap>        libs &lt;- unique(instlib)</span>\n",
       "<span style=white-space:pre-wrap>        for (l in libs) if (type == \"both\") </span>\n",
       "<span style=white-space:pre-wrap>            install.packages(update[instlib == l, \"Package\"], </span>\n",
       "<span style=white-space:pre-wrap>                l, repos = repos, method = method, ..., type = type)</span>\n",
       "<span style=white-space:pre-wrap>        else install.packages(update[instlib == l, \"Package\"], </span>\n",
       "<span style=white-space:pre-wrap>            l, contriburl = contriburl, method = method, available = available, </span>\n",
       "<span style=white-space:pre-wrap>            ..., type = type)</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "}</code></pre>"
      ],
      "text/latex": [
       "\\begin{minted}{r}\n",
       "function (lib.loc = NULL, repos = getOption(\"repos\"), contriburl = contrib.url(repos, \n",
       "    type), method, instlib = NULL, ask = TRUE, available = NULL, \n",
       "    oldPkgs = NULL, ..., checkBuilt = FALSE, type = getOption(\"pkgType\")) \n",
       "\\{\n",
       "    force(ask)\n",
       "    text.select <- function(old) \\{\n",
       "        update <- NULL\n",
       "        for (k in seq\\_len(nrow(old))) \\{\n",
       "            cat(old{[}k, \"Package\"{]}, \":\\textbackslash{}n\", \"Version\", old{[}k, \"Installed\"{]}, \n",
       "                \"installed in\", old{[}k, \"LibPath\"{]}, if (checkBuilt) \n",
       "                  paste(\"built under R\", old{[}k, \"Built\"{]}), \"\\textbackslash{}n\", \n",
       "                \"Version\", old{[}k, \"ReposVer\"{]}, \"available at\", \n",
       "                simplifyRepos(old{[}k, \"Repository\"{]}, type))\n",
       "            cat(\"\\textbackslash{}n\")\n",
       "            answer <- askYesNo(\"Update?\")\n",
       "            if (is.na(answer)) \\{\n",
       "                cat(\"cancelled by user\\textbackslash{}n\")\n",
       "                return(invisible())\n",
       "            \\}\n",
       "            if (isTRUE(answer)) \n",
       "                update <- rbind(update, old{[}k, {]})\n",
       "        \\}\n",
       "        update\n",
       "    \\}\n",
       "    if (is.null(lib.loc)) \n",
       "        lib.loc <- .libPaths()\n",
       "    if (type == \"both\" \\&\\& (!missing(contriburl) \\textbar{}\\textbar{} !is.null(available))) \\{\n",
       "        stop(\"specifying 'contriburl' or 'available' requires a single type, not type = \\textbackslash{}\"both\\textbackslash{}\"\")\n",
       "    \\}\n",
       "    if (is.null(available)) \\{\n",
       "        available <- available.packages(contriburl = contriburl, \n",
       "            method = method, ...)\n",
       "        if (missing(repos)) \n",
       "            repos <- getOption(\"repos\")\n",
       "    \\}\n",
       "    if (!is.matrix(oldPkgs) \\&\\& is.character(oldPkgs)) \\{\n",
       "        subset <- oldPkgs\n",
       "        oldPkgs <- NULL\n",
       "    \\}\n",
       "    else subset <- NULL\n",
       "    if (is.null(oldPkgs)) \\{\n",
       "        oldPkgs <- old.packages(lib.loc = lib.loc, contriburl = contriburl, \n",
       "            method = method, available = available, checkBuilt = checkBuilt)\n",
       "        if (missing(repos)) \n",
       "            repos <- getOption(\"repos\")\n",
       "        if (!is.null(oldPkgs)) \\{\n",
       "            pkg <- 0L\n",
       "            while (pkg < nrow(oldPkgs)) \\{\n",
       "                pkg <- pkg + 1L\n",
       "                if (find.package(oldPkgs{[}pkg{]}, lib.loc = lib.loc) != \n",
       "                  find.package(oldPkgs{[}pkg{]}, lib.loc = oldPkgs{[}pkg, \n",
       "                    2{]})) \\{\n",
       "                  warning(sprintf(\"package '\\%s' in library '\\%s' will not be updated\", \n",
       "                    oldPkgs{[}pkg{]}, oldPkgs{[}pkg, 2{]}), call. = FALSE, \n",
       "                    immediate. = TRUE)\n",
       "                  oldPkgs <- oldPkgs{[}-pkg, , drop = FALSE{]}\n",
       "                  pkg <- pkg - 1L\n",
       "                \\}\n",
       "            \\}\n",
       "        \\}\n",
       "        if (is.null(oldPkgs)) \n",
       "            return(invisible())\n",
       "    \\}\n",
       "    else if (!(is.matrix(oldPkgs) \\&\\& is.character(oldPkgs))) \n",
       "        stop(\"invalid 'oldPkgs'; must be a character vector or a result from old.packages()\")\n",
       "    if (!is.null(subset)) \\{\n",
       "        oldPkgs <- oldPkgs{[}rownames(oldPkgs) \\%in\\% subset, , drop = FALSE{]}\n",
       "        if (nrow(oldPkgs) == 0) \n",
       "            return(invisible())\n",
       "    \\}\n",
       "    update <- if (is.character(ask) \\&\\& ask == \"graphics\") \\{\n",
       "        if (.Platform\\$OS.type == \"windows\" \\textbar{}\\textbar{} .Platform\\$GUI == \n",
       "            \"AQUA\" \\textbar{}\\textbar{} (capabilities(\"tcltk\") \\&\\& capabilities(\"X11\"))) \\{\n",
       "            k <- select.list(oldPkgs{[}, 1L{]}, oldPkgs{[}, 1L{]}, multiple = TRUE, \n",
       "                title = \"Packages to be updated\", graphics = TRUE)\n",
       "            oldPkgs{[}match(k, oldPkgs{[}, 1L{]}), , drop = FALSE{]}\n",
       "        \\}\n",
       "        else text.select(oldPkgs)\n",
       "    \\}\n",
       "    else if (isTRUE(ask)) \n",
       "        text.select(oldPkgs)\n",
       "    else oldPkgs\n",
       "    if (length(update)) \\{\n",
       "        if (is.null(instlib)) \n",
       "            instlib <- update{[}, \"LibPath\"{]}\n",
       "        libs <- unique(instlib)\n",
       "        for (l in libs) if (type == \"both\") \n",
       "            install.packages(update{[}instlib == l, \"Package\"{]}, \n",
       "                l, repos = repos, method = method, ..., type = type)\n",
       "        else install.packages(update{[}instlib == l, \"Package\"{]}, \n",
       "            l, contriburl = contriburl, method = method, available = available, \n",
       "            ..., type = type)\n",
       "    \\}\n",
       "\\}\n",
       "\\end{minted}"
      ],
      "text/markdown": [
       "```r\n",
       "function (lib.loc = NULL, repos = getOption(\"repos\"), contriburl = contrib.url(repos, \n",
       "    type), method, instlib = NULL, ask = TRUE, available = NULL, \n",
       "    oldPkgs = NULL, ..., checkBuilt = FALSE, type = getOption(\"pkgType\")) \n",
       "{\n",
       "    force(ask)\n",
       "    text.select <- function(old) {\n",
       "        update <- NULL\n",
       "        for (k in seq_len(nrow(old))) {\n",
       "            cat(old[k, \"Package\"], \":\\n\", \"Version\", old[k, \"Installed\"], \n",
       "                \"installed in\", old[k, \"LibPath\"], if (checkBuilt) \n",
       "                  paste(\"built under R\", old[k, \"Built\"]), \"\\n\", \n",
       "                \"Version\", old[k, \"ReposVer\"], \"available at\", \n",
       "                simplifyRepos(old[k, \"Repository\"], type))\n",
       "            cat(\"\\n\")\n",
       "            answer <- askYesNo(\"Update?\")\n",
       "            if (is.na(answer)) {\n",
       "                cat(\"cancelled by user\\n\")\n",
       "                return(invisible())\n",
       "            }\n",
       "            if (isTRUE(answer)) \n",
       "                update <- rbind(update, old[k, ])\n",
       "        }\n",
       "        update\n",
       "    }\n",
       "    if (is.null(lib.loc)) \n",
       "        lib.loc <- .libPaths()\n",
       "    if (type == \"both\" && (!missing(contriburl) || !is.null(available))) {\n",
       "        stop(\"specifying 'contriburl' or 'available' requires a single type, not type = \\\"both\\\"\")\n",
       "    }\n",
       "    if (is.null(available)) {\n",
       "        available <- available.packages(contriburl = contriburl, \n",
       "            method = method, ...)\n",
       "        if (missing(repos)) \n",
       "            repos <- getOption(\"repos\")\n",
       "    }\n",
       "    if (!is.matrix(oldPkgs) && is.character(oldPkgs)) {\n",
       "        subset <- oldPkgs\n",
       "        oldPkgs <- NULL\n",
       "    }\n",
       "    else subset <- NULL\n",
       "    if (is.null(oldPkgs)) {\n",
       "        oldPkgs <- old.packages(lib.loc = lib.loc, contriburl = contriburl, \n",
       "            method = method, available = available, checkBuilt = checkBuilt)\n",
       "        if (missing(repos)) \n",
       "            repos <- getOption(\"repos\")\n",
       "        if (!is.null(oldPkgs)) {\n",
       "            pkg <- 0L\n",
       "            while (pkg < nrow(oldPkgs)) {\n",
       "                pkg <- pkg + 1L\n",
       "                if (find.package(oldPkgs[pkg], lib.loc = lib.loc) != \n",
       "                  find.package(oldPkgs[pkg], lib.loc = oldPkgs[pkg, \n",
       "                    2])) {\n",
       "                  warning(sprintf(\"package '%s' in library '%s' will not be updated\", \n",
       "                    oldPkgs[pkg], oldPkgs[pkg, 2]), call. = FALSE, \n",
       "                    immediate. = TRUE)\n",
       "                  oldPkgs <- oldPkgs[-pkg, , drop = FALSE]\n",
       "                  pkg <- pkg - 1L\n",
       "                }\n",
       "            }\n",
       "        }\n",
       "        if (is.null(oldPkgs)) \n",
       "            return(invisible())\n",
       "    }\n",
       "    else if (!(is.matrix(oldPkgs) && is.character(oldPkgs))) \n",
       "        stop(\"invalid 'oldPkgs'; must be a character vector or a result from old.packages()\")\n",
       "    if (!is.null(subset)) {\n",
       "        oldPkgs <- oldPkgs[rownames(oldPkgs) %in% subset, , drop = FALSE]\n",
       "        if (nrow(oldPkgs) == 0) \n",
       "            return(invisible())\n",
       "    }\n",
       "    update <- if (is.character(ask) && ask == \"graphics\") {\n",
       "        if (.Platform$OS.type == \"windows\" || .Platform$GUI == \n",
       "            \"AQUA\" || (capabilities(\"tcltk\") && capabilities(\"X11\"))) {\n",
       "            k <- select.list(oldPkgs[, 1L], oldPkgs[, 1L], multiple = TRUE, \n",
       "                title = \"Packages to be updated\", graphics = TRUE)\n",
       "            oldPkgs[match(k, oldPkgs[, 1L]), , drop = FALSE]\n",
       "        }\n",
       "        else text.select(oldPkgs)\n",
       "    }\n",
       "    else if (isTRUE(ask)) \n",
       "        text.select(oldPkgs)\n",
       "    else oldPkgs\n",
       "    if (length(update)) {\n",
       "        if (is.null(instlib)) \n",
       "            instlib <- update[, \"LibPath\"]\n",
       "        libs <- unique(instlib)\n",
       "        for (l in libs) if (type == \"both\") \n",
       "            install.packages(update[instlib == l, \"Package\"], \n",
       "                l, repos = repos, method = method, ..., type = type)\n",
       "        else install.packages(update[instlib == l, \"Package\"], \n",
       "            l, contriburl = contriburl, method = method, available = available, \n",
       "            ..., type = type)\n",
       "    }\n",
       "}\n",
       "```"
      ],
      "text/plain": [
       "function (lib.loc = NULL, repos = getOption(\"repos\"), contriburl = contrib.url(repos, \n",
       "    type), method, instlib = NULL, ask = TRUE, available = NULL, \n",
       "    oldPkgs = NULL, ..., checkBuilt = FALSE, type = getOption(\"pkgType\")) \n",
       "{\n",
       "    force(ask)\n",
       "    text.select <- function(old) {\n",
       "        update <- NULL\n",
       "        for (k in seq_len(nrow(old))) {\n",
       "            cat(old[k, \"Package\"], \":\\n\", \"Version\", old[k, \"Installed\"], \n",
       "                \"installed in\", old[k, \"LibPath\"], if (checkBuilt) \n",
       "                  paste(\"built under R\", old[k, \"Built\"]), \"\\n\", \n",
       "                \"Version\", old[k, \"ReposVer\"], \"available at\", \n",
       "                simplifyRepos(old[k, \"Repository\"], type))\n",
       "            cat(\"\\n\")\n",
       "            answer <- askYesNo(\"Update?\")\n",
       "            if (is.na(answer)) {\n",
       "                cat(\"cancelled by user\\n\")\n",
       "                return(invisible())\n",
       "            }\n",
       "            if (isTRUE(answer)) \n",
       "                update <- rbind(update, old[k, ])\n",
       "        }\n",
       "        update\n",
       "    }\n",
       "    if (is.null(lib.loc)) \n",
       "        lib.loc <- .libPaths()\n",
       "    if (type == \"both\" && (!missing(contriburl) || !is.null(available))) {\n",
       "        stop(\"specifying 'contriburl' or 'available' requires a single type, not type = \\\"both\\\"\")\n",
       "    }\n",
       "    if (is.null(available)) {\n",
       "        available <- available.packages(contriburl = contriburl, \n",
       "            method = method, ...)\n",
       "        if (missing(repos)) \n",
       "            repos <- getOption(\"repos\")\n",
       "    }\n",
       "    if (!is.matrix(oldPkgs) && is.character(oldPkgs)) {\n",
       "        subset <- oldPkgs\n",
       "        oldPkgs <- NULL\n",
       "    }\n",
       "    else subset <- NULL\n",
       "    if (is.null(oldPkgs)) {\n",
       "        oldPkgs <- old.packages(lib.loc = lib.loc, contriburl = contriburl, \n",
       "            method = method, available = available, checkBuilt = checkBuilt)\n",
       "        if (missing(repos)) \n",
       "            repos <- getOption(\"repos\")\n",
       "        if (!is.null(oldPkgs)) {\n",
       "            pkg <- 0L\n",
       "            while (pkg < nrow(oldPkgs)) {\n",
       "                pkg <- pkg + 1L\n",
       "                if (find.package(oldPkgs[pkg], lib.loc = lib.loc) != \n",
       "                  find.package(oldPkgs[pkg], lib.loc = oldPkgs[pkg, \n",
       "                    2])) {\n",
       "                  warning(sprintf(\"package '%s' in library '%s' will not be updated\", \n",
       "                    oldPkgs[pkg], oldPkgs[pkg, 2]), call. = FALSE, \n",
       "                    immediate. = TRUE)\n",
       "                  oldPkgs <- oldPkgs[-pkg, , drop = FALSE]\n",
       "                  pkg <- pkg - 1L\n",
       "                }\n",
       "            }\n",
       "        }\n",
       "        if (is.null(oldPkgs)) \n",
       "            return(invisible())\n",
       "    }\n",
       "    else if (!(is.matrix(oldPkgs) && is.character(oldPkgs))) \n",
       "        stop(\"invalid 'oldPkgs'; must be a character vector or a result from old.packages()\")\n",
       "    if (!is.null(subset)) {\n",
       "        oldPkgs <- oldPkgs[rownames(oldPkgs) %in% subset, , drop = FALSE]\n",
       "        if (nrow(oldPkgs) == 0) \n",
       "            return(invisible())\n",
       "    }\n",
       "    update <- if (is.character(ask) && ask == \"graphics\") {\n",
       "        if (.Platform$OS.type == \"windows\" || .Platform$GUI == \n",
       "            \"AQUA\" || (capabilities(\"tcltk\") && capabilities(\"X11\"))) {\n",
       "            k <- select.list(oldPkgs[, 1L], oldPkgs[, 1L], multiple = TRUE, \n",
       "                title = \"Packages to be updated\", graphics = TRUE)\n",
       "            oldPkgs[match(k, oldPkgs[, 1L]), , drop = FALSE]\n",
       "        }\n",
       "        else text.select(oldPkgs)\n",
       "    }\n",
       "    else if (isTRUE(ask)) \n",
       "        text.select(oldPkgs)\n",
       "    else oldPkgs\n",
       "    if (length(update)) {\n",
       "        if (is.null(instlib)) \n",
       "            instlib <- update[, \"LibPath\"]\n",
       "        libs <- unique(instlib)\n",
       "        for (l in libs) if (type == \"both\") \n",
       "            install.packages(update[instlib == l, \"Package\"], \n",
       "                l, repos = repos, method = method, ..., type = type)\n",
       "        else install.packages(update[instlib == l, \"Package\"], \n",
       "            l, contriburl = contriburl, method = method, available = available, \n",
       "            ..., type = type)\n",
       "    }\n",
       "}\n",
       "<bytecode: 0x55a374502c98>\n",
       "<environment: namespace:utils>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "update.packages"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "149c3889-637f-4a58-bd44-7e58df057104",
   "metadata": {},
   "source": [
    "Chargement de minfi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "12da158e-4071-4b07-8fb6-2dd208a6fbeb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: bumphunter\n",
      "\n",
      "Loading required package: foreach\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: locfit\n",
      "\n",
      "locfit 1.5-9.4 \t 2020-03-24\n",
      "\n",
      "Possible Ensembl SSL connectivity problems detected.\n",
      "Please see the 'Connection Troubleshooting' section of the biomaRt vignette\n",
      "vignette('accessing_ensembl', package = 'biomaRt')Error in curl::curl_fetch_memory(url, handle = handle) : \n",
      "  SSL certificate problem: certificate has expired\n",
      "\n",
      "\n",
      "Setting options('download.file.method.GEOquery'='auto')\n",
      "\n",
      "Setting options('GEOquery.inmemory.gpl'=FALSE)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(minfi)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d180dbe-1aa6-412a-ac07-8115056505e8",
   "metadata": {},
   "source": [
    "**Récupération des data à partir de GEOquery**\n",
    "charger GEOquery pour importer les datas publiques d'une étude "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ea9b6575-7e7e-4842-9422-f08ab6812bdf",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(GEOquery)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4070751d-2aa6-4475-a154-0436e0f65e96",
   "metadata": {},
   "source": [
    "si données dans supplemental files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e4ef3de8-f957-4075-ae9a-0eb7e4c450d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>size</th><th scope=col>isdir</th><th scope=col>mode</th><th scope=col>mtime</th><th scope=col>ctime</th><th scope=col>atime</th><th scope=col>uid</th><th scope=col>gid</th><th scope=col>uname</th><th scope=col>grname</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;octmode&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar</th><td>472432640</td><td>FALSE</td><td>660</td><td>2023-11-13 12:18:28</td><td>2023-11-13 12:18:28</td><td>2023-11-10 16:33:42</td><td>163697</td><td>163697</td><td>sletuve</td><td>sletuve</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & size & isdir & mode & mtime & ctime & atime & uid & gid & uname & grname\\\\\n",
       "  & <dbl> & <lgl> & <octmode> & <dttm> & <dttm> & <dttm> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t/shared/ifbstor1/projects/202304\\_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716\\_RAW.tar & 472432640 & FALSE & 660 & 2023-11-13 12:18:28 & 2023-11-13 12:18:28 & 2023-11-10 16:33:42 & 163697 & 163697 & sletuve & sletuve\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 10\n",
       "\n",
       "| <!--/--> | size &lt;dbl&gt; | isdir &lt;lgl&gt; | mode &lt;octmode&gt; | mtime &lt;dttm&gt; | ctime &lt;dttm&gt; | atime &lt;dttm&gt; | uid &lt;int&gt; | gid &lt;int&gt; | uname &lt;chr&gt; | grname &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar | 472432640 | FALSE | 660 | 2023-11-13 12:18:28 | 2023-11-13 12:18:28 | 2023-11-10 16:33:42 | 163697 | 163697 | sletuve | sletuve |\n",
       "\n"
      ],
      "text/plain": [
       "                                                                                                   size     \n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 472432640\n",
       "                                                                                                   isdir\n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar FALSE\n",
       "                                                                                                   mode\n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 660 \n",
       "                                                                                                   mtime              \n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 2023-11-13 12:18:28\n",
       "                                                                                                   ctime              \n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 2023-11-13 12:18:28\n",
       "                                                                                                   atime              \n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 2023-11-10 16:33:42\n",
       "                                                                                                   uid   \n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 163697\n",
       "                                                                                                   gid   \n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 163697\n",
       "                                                                                                   uname  \n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar sletuve\n",
       "                                                                                                   grname \n",
       "/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar sletuve"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getGEOSuppFiles(\"GSE137716\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ff6f389-506d-43d7-b273-5f320a598ed5",
   "metadata": {},
   "source": [
    "décompresser archive qui contient les idat files (_RAW) et extraire dans dossier exdir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2432a84a-d004-4370-ab8a-3f08e521f30b",
   "metadata": {},
   "outputs": [],
   "source": [
    "untar(\"GSE137716/GSE137716_RAW.tar\", exdir = \"GSE137716/idat\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "387dc3d0-28a5-4f16-bbb6-cb94351bdba5",
   "metadata": {},
   "source": [
    "On voit ici qu'il y a des files Grn (green) et Red, 2 files pour chaque échantillon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4e7e5fa9-23e2-4484-8ada-3dc111024459",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GSM4085551_9976861129_R03C02_Grn.idat.gz'</li><li>'GSM4085551_9976861129_R03C02_Red.idat.gz'</li><li>'GSM4085552_9979858039_R01C01_Grn.idat.gz'</li><li>'GSM4085552_9979858039_R01C01_Red.idat.gz'</li><li>'GSM4085553_9976861137_R03C02_Grn.idat.gz'</li><li>'GSM4085553_9976861137_R03C02_Red.idat.gz'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GSM4085551\\_9976861129\\_R03C02\\_Grn.idat.gz'\n",
       "\\item 'GSM4085551\\_9976861129\\_R03C02\\_Red.idat.gz'\n",
       "\\item 'GSM4085552\\_9979858039\\_R01C01\\_Grn.idat.gz'\n",
       "\\item 'GSM4085552\\_9979858039\\_R01C01\\_Red.idat.gz'\n",
       "\\item 'GSM4085553\\_9976861137\\_R03C02\\_Grn.idat.gz'\n",
       "\\item 'GSM4085553\\_9976861137\\_R03C02\\_Red.idat.gz'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GSM4085551_9976861129_R03C02_Grn.idat.gz'\n",
       "2. 'GSM4085551_9976861129_R03C02_Red.idat.gz'\n",
       "3. 'GSM4085552_9979858039_R01C01_Grn.idat.gz'\n",
       "4. 'GSM4085552_9979858039_R01C01_Red.idat.gz'\n",
       "5. 'GSM4085553_9976861137_R03C02_Grn.idat.gz'\n",
       "6. 'GSM4085553_9976861137_R03C02_Red.idat.gz'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GSM4085551_9976861129_R03C02_Grn.idat.gz\"\n",
       "[2] \"GSM4085551_9976861129_R03C02_Red.idat.gz\"\n",
       "[3] \"GSM4085552_9979858039_R01C01_Grn.idat.gz\"\n",
       "[4] \"GSM4085552_9979858039_R01C01_Red.idat.gz\"\n",
       "[5] \"GSM4085553_9976861137_R03C02_Grn.idat.gz\"\n",
       "[6] \"GSM4085553_9976861137_R03C02_Red.idat.gz\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(list.files(\"GSE137716/idat\", pattern = \"idat\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12ce09d6-e119-437d-9cb5-6b3f2cda2ce5",
   "metadata": {},
   "source": [
    "décompresser les idat files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "212d33b3-e845-4a1f-9ae1-c77a7a97dc05",
   "metadata": {},
   "outputs": [],
   "source": [
    "idatFiles <- list.files(\"GSE68777/idat\", pattern = \"idat.gz$\", full = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d98572b5-141b-47a0-8b64-245d83c44757",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "\n",
       "\n"
      ],
      "text/plain": [
       "named list()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sapply(idatFiles, gunzip, overwrite = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76453c55-ca65-4e93-84d8-e19e50fa5414",
   "metadata": {},
   "source": [
    "Lire les files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "744f4b98-7da8-4025-9ae7-09688c83aa72",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in read.metharray.exp(\"GSE137716/idat\"): No IDAT files were found\n",
     "output_type": "error",
     "traceback": [
      "Error in read.metharray.exp(\"GSE137716/idat\"): No IDAT files were found\nTraceback:\n",
      "1. read.metharray.exp(\"GSE137716/idat\")",
      "2. stop(\"No IDAT files were found\")"
     ]
    }
   ],
   "source": [
    "rgSet <- read.metharray.exp(\"GSE137716/idat\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6d070ab4-245d-4b1d-8213-ff028f0e9574",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.0.3",
   "language": "R",
   "name": "r-4.0.3"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
